doc: release: 3.7: Add release notes for Flash (#75041)
zephyrproject-rtos/zephyr: doc/releases/release-notes-3.7.rst

* Flash

  * Added support for Ambiq Apollo3 series.

>   * Added support for multi-instance of SPI NOR driver (spi_nor.c).

Reviewer: ```suggestion
  * Added support for multiple instances of the SPI NOR driver (spi_nor.c).
```
maybe better?

  * Added preliminary support for non-erase devices with introduction of

>     device capabilities to c:struct:`flash_parameters` and utility function
>     c:func:`flash_params_get_erase_cap` that allows to obtain erase type
>     provided by a device; added c:macro:`FLASH_ERASE_C_EXPLICIT`, which is
>     currently only supported erase type and is set by all flash devices.

Reviewer: ```suggestion
    device capabilities to c:struct:`flash_parameters` and the utility function
    c:func:`flash_params_get_erase_cap` that allows to obtain the erase type
    provided by a device; added c:macro:`FLASH_ERASE_C_EXPLICIT`, which is
    currently the only supported erase type and is set by all flash devices.
```
maybe better?

>   * Added c:func:`flash_flatten` function that can be used on devices

Reviewer: ```suggestion
  * Added the c:func:`flash_flatten` function that can be used on devices
```

>     with and without erase when erase is used not for preparing device for

Reviewer: If there something missing before that "when"?
(I can't really parse this sentence)

Author: Improved, hopefully, the description.

    random data write but rather to remove/scramble device.

>   * Added c:func:`flash_fill` utility function that allows to write
>     single value across provided range of selected device.

Reviewer: ```suggestion
  * Added the c:func:`flash_fill` utility function which allows to write
    a single value across a provided range in a selected device.
```
